api/envoy/extensions/filters/http/mutation/v3alpha/processing_mode.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.mutation.v3alpha;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.mutation.v3alpha";
+option java_outer_classname = "ProcessingModeProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Mutation Service]
+// Mutation Service Processing Mode
+// [#extension: envoy.filters.http.mutation]
+
+// This configuration describes how a mutation filter processes the HTTP
+// request or response -- specifically, whether it is done synchronously
+// or asynchronously.
+
+message ProcessingMode {
+  enum Mode {
+    // In the default "sync" mode, a message is sent to the filter server
+    // as soon as decodeHeaders is called, and the filter waits for a response
+    // from the server before continuing.
+    SYNC = 0;
+
+    // In "async" mode, messages are sent to the filter server and the filter
+    // does not wait for a response. By default the request and response headers,
+    // all body chunks, and trailers are sent. An AsyncParams object can be used
+    // to customize this behavior.
+    ASYNC = 1;
+  }
+
+  Mode mode = 1;
+
+  // The optional AsyncParams object controls how data is delivered
+  // to the server in ASYNC mode. It is ignored in SYNC mode.
```

Comment:
Why is it ignored in SYNC mode? Perhaps for performance reasons the user only wants to send over headers and mutate them and then stream everything else? I think it would be more flexible to rename this `Params` or `DataConfiguration` or something and use in both modes?